tests/src/deque/src/main.sw

```diff
@@ -0,0 +1,37 @@
+script;
+
+use deque::Deque;
+use std::assert::assert;
+
+fn main() -> bool {
+    let mut deque_push_front = Deque::new();
+    deque_push_front.push_front(1);
```

Comment:
The `push_front()` test can be written as follows

1. Create a new Deque
2. Assert that the instance has no items
3. Push to the front once
4. Assert that the instance has increased in length by 1 and the item in the instance has the value that you have added
5. Push once more
6. Assert again that the length has increased by 1 and now assert that the first item and second item both have the correct values.

This ensures that each call to your function behaves as expected and there are no side effects after X number of calls.

We can be repeated for other test cases.